core/src/main/java/io/temporal/samples/hello/HelloCancellationScopeWithTimer.java

```diff
@@ -0,0 +1,231 @@
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.hello;
+
+import io.temporal.activity.*;
+import io.temporal.client.ActivityCompletionException;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.failure.ActivityFailure;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.*;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+public class HelloCancellationScopeWithTimer {
+  // Define the task queue name
+  static final String TASK_QUEUE = "HelloCancellationScopeTimerTaskQueue";
+
+  // Define our workflow unique id
+  static final String WORKFLOW_ID = "HelloCancellationScopeTimerWorkflow";
+
+  /**
+   * The Workflow Definition's Interface must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow Definitions should not contain any heavyweight computations, non-deterministic
+   * code, network calls, database operations, etc. Those things should be handled by the
+   * Activities.
+   *
+   * @see io.temporal.workflow.WorkflowInterface
+   * @see io.temporal.workflow.WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface CancellationWithTimerWorkflow {
+
+    /**
+     * This is the method that is executed when the Workflow Execution is started. The Workflow
+     * Execution completes when this method finishes execution.
+     */
+    @WorkflowMethod
+    String execute(String input);
+  }
+
+  /**
+   * This is the Activity Definition's Interface. Activities are building blocks of any Temporal
+   * Workflow and contain any business logic that could perform long running computation, network
+   * calls, etc.
+   *
+   * <p>Annotating Activity Definition methods with @ActivityMethod is optional.
+   *
+   * @see io.temporal.activity.ActivityInterface
+   * @see io.temporal.activity.ActivityMethod
+   */
+  @ActivityInterface
+  public interface UpdateInfoActivities {
+    String updateInfo(String input);
+  }
+
+  // Define the workflow implementation which implements our getGreeting workflow method.
+  public static class CancellationWithTimerWorkflowImpl implements CancellationWithTimerWorkflow {
+    private final UpdateInfoActivities activities =
+        Workflow.newActivityStub(
+            UpdateInfoActivities.class,
+            ActivityOptions.newBuilder()
+                // If heartbeat timeout is not set, activity heartbeats will be throttled to one
+                // every 30 seconds
+                .setHeartbeatTimeout(Duration.ofSeconds(2))
```

Comment:
The activity also just won't have a heartbeat timeout

Reply:
ugh...forgot. fixing

Reply:
thanks. updated